### PR TITLE
Form

### LINKED ISSUE #22: Feature: Add GameDisputeForm component to [gameid] page
- Includes:

### PR DESCRIPTION
**Pull request by krogowsk531**
# Team members present: Hanna, Michael, Katy

## What's this PR do? 

Testing, Form Component, Start of Hooks for local storage

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Styling
- [x] Testing
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Where should the reviewer start? 

Can see new form component added to single game page

## How should this be manually tested? 

Navigate to single game page

## Any background context you want to provide? 

Trying to find a way to allow comments and such to persist, Still having issues getting tests to pass when css module is used

## What are the relevant tickets? 
* Addresses #3
* Closes #22

## Screenshots (if appropriate) 

* See screenshots in stretch project channel addressed to Khalid

### CODE REVIEW

File: components/Cards/Cards.js [hunk 1]
```diff
@@ -14,12 +15,13 @@ const Cards = (props) => {
     })
 
 return (
-    <div>  
+    <div>
     Cards
     { formattedCards }
+
```
Comment (MichaelEWalker87) [inline components/Cards/Cards.js:21]: I think we can remove this

File: components/NavBar/NavBar.test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { render, screen } from "@testing-library/react";
+import NavBar from "./NavBar"
+
+
+describe("NavBar", () => {
```
Comment (MichaelEWalker87) [inline components/NavBar/NavBar.test.js:5]: Do we need to test the routing in this or not?

File: components/RuleDisputeForm/RuleDisputeForm.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+//     <div>
+//       <input
+//         type="text"
+//         placeholder="Enter your name"
+//         value={name}
+//         onChange={e => setName(e.target.value)}
+//       />
+//     </div>
+//   );
+// }
```
Comment (MichaelEWalker87) [inline components/RuleDisputeForm/RuleDisputeForm.js:18]: I think we can remove this now

File: components/RuleDisputeForm/RuleDisputeForm.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        type='text'
+        name='officialRule'
+        />
+        <label>Interpretation of Rule: </label>
+        <input
+        type='text'
+        name='InterpretationRule'
+        />
+
+        <button>SUBMIT</button>
```
Comment (MichaelEWalker87) [inline components/RuleDisputeForm/RuleDisputeForm.js:58]: Do we want to keep SUBMIT or go with Submit or as we have in the design Start Fight?

**Review by hannakim91 — APPROVED** [review]
Glad we got a start to testing and learned that components are/behave as regular React components moving forward.

- Having a mentor help us figure out webpack configuration for using jest to test nextjs css modules
- Getting more familiar with hooks today before finishing the Form component

File: components/RuleDisputeForm/RuleDisputeForm.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  const [storedValue, setStoredValue] = useState(() => {
+    try {
+      const item = window.localStorage.getItem(key)
+      return item ? JSON.parse(item): initialValue
+    } catch (error) {
+      console.log(error)
+      return initialValue;
+    }
+  })
+}
```
Comment (MichaelEWalker87) [inline components/RuleDisputeForm/RuleDisputeForm.js:89]: This is rather interesting I would like to know more about this even though we had already looked up this

**Review by MichaelEWalker87 — APPROVED** [review]

**Pull request merged**